source/Glimpse.Ado/AlternateType/GlimpseDbProviderFactory.cs

```diff
@@ -53,7 +78,12 @@ public override DbConnectionStringBuilder CreateConnectionStringBuilder()
 
         public override DbDataAdapter CreateDataAdapter()
         {
-            return new GlimpseDbDataAdapter(InnerFactory.CreateDataAdapter()); 
+            var adapter = InnerFactory.CreateDataAdapter();
+            if (ShouldNotWrap())
+            {
+                return adapter;
+            }
+            return new GlimpseDbDataAdapter(adapter);
         }
 
```

Comment:
Maybe invert the check? As in `IsAdoMonitoringNeeded()` ? and then wrap if needed?